### ISSUE TITLE
Use MSBuild in process instead of shelling out

### ISSUE THREAD

**Issue by rynowak** [dotnet/tye#167]
One of the changes that I'm currently working on is refactoring and unifying our processing pipeline - connecting the m8s code and the opulence code that are still pretty separate. As part of this change, it would be nice if we could get all of the data we want up front - we need the project-related info for deployment and it would be nice to have it for local dev features as well.

I was concerned about the impact this would have on startup speed. We currently shell out to `dotnet msbuild` at the command line to exfiltrate data from the project, and it's slow. Adding more than 1-2 seconds of time to `tye run` would make me really concerned.

I coded up an approach for doing evaluation inside the process, and it's much better 👍 and a lil less complicated. (I'm calling it evaluation, but it's really not, because we need to run some targets to do things like resolve shared frameworks and versions).

**Example: (three projects)**

```txt
Finding MSBuild took: 144.5537ms
Reading Project Details...
    Evaluation Took: 865.8865ms
    dotnet msbuild Took: 951.8802ms
Reading Project Details...
    Evaluation Took: 233.8668ms
    dotnet msbuild Took: 948.8874ms
Reading Project Details...
    Evaluation Took: 170.6006ms
    dotnet msbuild Took: 890.1061ms
```

The approach of running MSBuild in process adds a fixed ~150ms cost (see https://github.com/microsoft/MSBuildLocator) to find MSBuild and load the MSBuild binaries from the .NET SDK instead of whatever random dependencies we have. 

For demonstration purposes this example is using an in-process evaluation and then running the `dotnet msbuild` technique back to back.

So by the time you've got two services, doing stuff in process is a clear performance win.

**Comment by rynowak** [comment]
@rainersigwald for potential interest and feedback 👍

**Comment by rynowak** [comment]
> Adding more than 1-2 seconds of time to `tye run` would make me really concerned.

And re: this - we can probably make up this time once we get to this approach by building in-process, daemonizing, and doing stuff in parallel. Basically once we lock on to doing stuff in memory, we have lots of ways to amortize other costs we currently have - for the 1 sec or so of fixed cost and ~200ms per project.
